app/main/views/search.py

```diff
@@ -59,6 +59,20 @@ def create(index_name):
 
 @main.route('/<string:index_name>', methods=['DELETE'])
 def delete(index_name):
+    status, status_code = status_for_index(index_name)
+
+    if status_code != 200:
```

Comment:
That's a good question. The only one I've seen is the 404 on the missing index, which returns the exact same error message as the deletion attempt would. Other than that I couldn't find a list of possible responses in the documentation , so I'm not really sure. 

I only need this condition to simplify the "if the key is missing it's an alias" step, I could try to get around it without checking the response status. Or we could leave this as is and see if it becomes a problem down the line. As you've said, the only thing it might do is prevent us from deleting something we might want to delete.